src/sysdolphin/baselib/memory.c

```diff
@@ -1,16 +1,18 @@
+#include "Runtime/platform.h"
 #include <sysdolphin/baselib/memory.h>
 
 #include <dolphin/os/OSAlloc.h>
 #include <sysdolphin/baselib/debug.h>
+#include <sysdolphin/baselib/initialize.h>
 
-void HSD_Free(void* ptr)
+void HSD_Free(any_t ptr)
 {
     OSFreeToHeap(HSD_GetHeap(), ptr);
 }
 
-void* HSD_MemAlloc(s32 size)
+any_t HSD_MemAlloc(ptrdiff_t size)
```

Comment:
```suggestion
any_t HSD_MemAlloc(ssize_t size)
```

Should definitely be `size_t` but doesn't match because it needs to be signed. Let's add a `ssize_t` typedef, since that's probably the closest to correct.

Reply:
That's literally what [`ptrdiff_t`](https://en.cppreference.com/w/cpp/types/ptrdiff_t) is as far as I understand it.

Reply:
They're pretty much always the same underlying type, but a ptrdiff is a difference of pointers and a size_t counts a number of bytes, so it makes more sense to pass a ssize_t to an allocator IMO